Remove preprocessor in favor of secondary inputs

andresmgot opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

7. If your PR content should be added to the What's New document for the next major or minor release, add the **add to what's new** label to your PR. Note that you should add this label to the main PR that introduces the feature; do not add this label to smaller PRs for the feature.

-->

**What is this feature?**

Rather than unsing `prepocessor` in our API (which is not present in the GCM API), use the `secondary.*` inputs.

Note that the logic to migrate a preprocessor to the related inputs should still somewhere (because the users won't manually enter them). My preferred place for this would be the frontend, since it's a UI abstraction, but since this code needs to also be in the backend for migration purposes we may decide to leave it just there.

**Which issue(s) does this PR fix?**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Ref #58333

**Special notes for your reviewer**:

andresmgot committed:
Remove preprocessor in favor of secondary inputs

asimpson reviewed (approved):
:trophy: 

Nice work here. I dig having a defined type for `timeSeriesList` instead of an interface along with the additional properties you added. is there a better place the `aggregation` rules could live instead of #30866? Is it worth it to move them somewhere else?

Tests pass for me locally as well.

andresmgot committed:
review

andresmgot commented:
> is there a better place the `aggregation` rules could live instead of #30866? Is it worth it to move them somewhere else?

I have been looking for some official docs from google related to that and it's nowhere :shrug: 

I have modified the code a bit and added come comments so the three steps to perform are clear (not having to reach the external link).
